src/main/flight/pid.c

```diff
@@ -1130,6 +1186,15 @@ static void pidApplyYawMode3(void)
                             pid.data[axis].F + pid.data[axis].B;
 }
 
+void pidCalcOffsetFloodRelaxFactor(const pidProfile_t *profile)
+{
+    const float collective = getCollectiveDeflection();
+    const float collectiveLpf =
+        pt1FilterApply(&pid.offsetFloodRelaxFilter, collective);
+    const float collectiveHpf = collective - collectiveLpf;
+    pid.offsetFloodRelaxFactor =
+        MAX(0, 1.0f - fabsf(collectiveHpf) / profile->offset_flood_relax_level);
```

Comment:
This function should be part of pidApplyOffsetFlood()

Reply:
Done